### PR TITLE
Don't clone Value when serializing

### PR DESCRIPTION
**Pull request by cmyr**
This is a sketch broken out from the discussion in #61.

The only tricky part here is `Event`, which needs to be owned when deserializing (not technically in the XML case, but practically, since `Value` is owned) but which we want to have be borrowed when serializing. I've got this working with a `Cow`; the slightly icky bit is that I need to explicitly specify that the `Cow` has a `'static` lifetime in the deserializing code.

That said, I think it's a reasonable solution?

If this is interesting to you I'd be happy to touch up some docs etc and.

I also haven't actually profiled the difference, although it should be measurable? I'd do that before wanting something like this merged.

### CODE REVIEW

File: src/value.rs [hunk 1]
```diff
@@ -96,20 +96,20 @@ impl Value {
...
     #[cfg(feature = "enable_unstable_features_that_may_break_with_minor_version_bumps")]
-    pub fn into_events(self) -> IntoEvents {
+    pub fn into_events(&self) -> IntoEvents {
         IntoEvents::new(self)
     }
 
     /// Converts a `Value` into an `Event` iterator.
     #[cfg(not(feature = "enable_unstable_features_that_may_break_with_minor_version_bumps"))]
-    pub(crate) fn into_events(self) -> IntoEvents {
+    pub(crate) fn into_events(&self) -> IntoEvents {
```
Comment (cmyr) [inline src/value.rs:112]: ultimately we would rename this to `to_events` or something similar, but I was trying to keep the changes minimal.

**Comment by ebarnard** [comment]
Looks like a good approach.

I'd be tempted to use `type OwnedEvent = Event<'static>` for readability but that's not very important.

I can't imagine this will have a huge effect on performance. A slight increase in the size of `Event` and a few more branches, but that's about it. Saying that, it would be nice if `plist` had a few benchmarks.

**Comment by cmyr** [comment]
Alright, I've added the `OwnedEvent` alias and did a bit of cleanup.

I renamed `into_events` to `to_events`, but it might be that just `events` is a better name? Do you have any opinions?

File: src/stream/mod.rs [hunk 2]
```diff
@@ -162,9 +181,9 @@ impl<R: Read + Seek> Reader<R> {
 }
 
 impl<R: Read + Seek> Iterator for Reader<R> {
-    type Item = Result<Event, Error>;
+    type Item = Result<Event<'static>, Error>;
```
Comment (ebarnard) [inline src/stream/mod.rs:184]: `OwnedEvent` (and below)

**Comment by ebarnard** [comment]
I think the names `events()` and `Events<'a>` are the most in keeping with Rust's stdlib naming conventions.

**Comment by cmyr** [comment]
Nope I think those are better names, I've made that change. This is hopefully ready to go?

**Comment by ebarnard** [comment]
It is, thanks.

**Comment by ebarnard** [comment]
Released in 1.2.0

**Pull request merged**